src/modules/Dashboard/features/Applet/Schedule/ActivityForm/TimersTab/TimersTab.tsx

```diff
@@ -1,46 +1,60 @@
-import { useState } from 'react';
 import { useTranslation } from 'react-i18next';
 import { useFormContext } from 'react-hook-form';
 
 import { TimePicker, ToggleButtonGroup } from 'shared/components';
-import theme from 'shared/styles/theme';
-import { StyledBodyLarge } from 'shared/styles/styledComponents';
+import { StyledBodyLarge, StyledBodyMedium, theme } from 'shared/styles';
+import { SelectController } from 'shared/components/FormComponents';
+import { TimerType } from 'modules/Dashboard/api';
 
-import { Timers, timersButtons } from './TimersTab.const.';
-import { FormValues } from '../';
+import { timersButtons, idleTimeOptions } from './TimersTab.const.';
+import { FormValues } from '../ActivityForm.types';
 
 export const TimersTab = () => {
   const { t } = useTranslation('app');
-  const { setValue } = useFormContext<FormValues>();
-  const [activeTimer, setActiveTimer] = useState<string>(Timers.NoTimeLimit);
+  const { watch, setValue, control } = useFormContext<FormValues>();
+  const activeTimer = watch('timerType');
 
-  const updateTimers = () => {
-    setValue('timerDuration', '');
-    setValue('idleTime', '');
-  };
+  const handleSetTimerType = (timerType: string) => setValue('timerType', timerType as TimerType);
 
   return (
     <>
+      <StyledBodyMedium sx={{ mb: theme.spacing(1.2) }}>{t('setTimeLimit')}</StyledBodyMedium>
       <ToggleButtonGroup
         toggleButtons={timersButtons}
         activeButton={activeTimer}
-        setActiveButton={setActiveTimer}
-        customChange={updateTimers}
+        setActiveButton={handleSetTimerType}
       />
-      {activeTimer === Timers.Timer && (
+      {activeTimer === TimerType.Timer && (
         <>
           <StyledBodyLarge sx={{ margin: theme.spacing(2.4, 0) }}>
             {t('timeToCompleteActivity')}
           </StyledBodyLarge>
           <TimePicker name="timerDuration" label={t('duration')} />
         </>
       )}
-      {activeTimer === Timers.IdleTime && (
+      {activeTimer === TimerType.Idle && (
         <>
           <StyledBodyLarge sx={{ margin: theme.spacing(2.4, 0) }}>
             {t('maximumTimeAwayFromActivity')}
           </StyledBodyLarge>
-          <TimePicker format="mm" name="idleTime" label={t('duration')} />
+          <SelectController
+            control={control}
+            name="idleTime"
+            label={t('duration')}
+            options={idleTimeOptions}
+            fullWidth
+            SelectProps={{
+              MenuProps: {
+                PaperProps: {
+                  sx: {
+                    maxHeight: '20rem',
+                  },
+                },
+              },
+            }}
+            customLiStyles={{ padding: `${theme.spacing(1, 1.6)} !important` }}
```

Comment:
I think it can be done without important

Reply:
a lot of efforts to do this, let's be as it is